tests/agents/test_set_physics_state.py

```diff
@@ -0,0 +1,139 @@
+import holodeck
+import uuid
+import numpy as np
+from holodeck import packagemanager as pm
+from holodeck.environments import HolodeckEnvironment
+from tests.utils.equality import almost_equal
+
+sphere_config = {
+    "name": "test_location_sensor",
+    "world": "TestWorld",
+    "main_agent": "turtle0",
+    "agents": [
+        {
+            "agent_name": "turtle0",
+            "agent_type": "TurtleAgent",
+            "sensors": [
+                {
+                    "sensor_type": "RotationSensor",
+                },
+                {
+                    "sensor_type": "LocationSensor",
+                },
+                {
+                    "sensor_type": "VelocitySensor",
+                },
+            ],
+            "control_scheme": 0,
+            "location": [.95, -1.75, .5]
+        }
+    ]
+}
+
+
+def test_set_physics_state_loc_and_rot():
+    """Validates that the set_physics_state function correctly sets the location and rotation of the agent.
+
+    """
+
+    binary_path = holodeck.packagemanager.get_binary_path_for_package("DefaultWorlds")
+
+    with holodeck.environments.HolodeckEnvironment(scenario=sphere_config,
```

Comment:
It seems like you could probably use the same environment for all of the tests in this file and just reset it before every use. There are two reasons you'd want to do this. The first is that it should make your tests run somewhat faster because you only have to initialize once, and the second is that we've run into a lot of thorny hard-to-debug issues that we think might have to do with having too many instances open at one time or starting and stopping instances in quick succession.

I'd recommend adapting what we did [here](https://github.com/BYU-PCCL/holodeck/blob/develop/tests/sensors/test_rgb_camera.py#L69-L96) for your use case. It uses a [PyTest fixture](https://docs.pytest.org/en/stable/fixture.html#fixtures-a-prime-example-of-dependency-injection) to allow scoping (caching) data over multiple tests. You could just stick that at the bottom of this file with a few changes:
- Change names (`shared_ticks_per_capture_env`, `ticks_per_capture_env`) to something more appropriate.
- Switch out the environment setup with your own using their pattern (you can ignore base_cfg and cfg)
- Add the name you set for the function as an argument to all of your tests (`def test_set_physics_state_vel(set_physics_state_env)` for example) and use it instead, removing the `with` statements.

Hopefully that makes sense. Let me know if doesn't.

Reply:
I made these changes, let me know if I did not do it correctly.